spec/requests/rights_flow_spec.rb

```diff
@@ -47,6 +47,55 @@
     expect(last_subscription.rights_hash)
       .to include('C-Talk with DDA before a Plea Agreement' => false)
 
-    # TODO: add assertions about the user creation
+    expect(last_subscription.email)
+      .to eq('tom@example.com')
+  end
+
+  it 'allows hitting back and saving an updated version' do
+    get '/rights'
+    follow_redirect!
+
+    post '/rights/who_assert', params: { rights_flow: { 'flag_a_assert_dda' => '1' } }
+    follow_redirect!
+
+    post '/rights/to_notification', params: { rights_flow: { 'flag_b_critical_stage' => '1' } }
+    follow_redirect!
+
+    post '/rights/to_financial_assistance', params: { rights_flow: { 'flag_k_restitution' => '1' } }
+    follow_redirect!
+
+    post '/rights/in_special_cases'
+    follow_redirect!
+
+    post '/rights/create_account', params: {
+      rights_flow: {
+        'first_name' => 'Tom',
+        'last_name' => 'Example',
+        'email' => 'tom@example.com',
+        'phone_number' => '330 555 1234',
+        'case_number' => '17CR1234'
+      }
+    }
+    follow_redirect!
+    expect(response.body).to include('all set')
+
+    post '/rights/create_account', params: {
+      rights_flow: {
+        'first_name' => 'Thomas',
+        'last_name' => 'Example',
+        'email' => 'Thomas@example.com',
+        'phone_number' => '330 123 1234',
+        'case_number' => '18CR1234'
+      }
+    }
+    follow_redirect!
+    expect(response.body).to include('Changes saved')
+
+    subscription = CourtCaseSubscription.find_by(
+      case_number: '18CR1234',
+      email: 'Thomas@example.com'
+    )
+    expect(subscription).to be_present
+    expect(subscription.phone_number).to eq('330 123 1234')
   end
 end
```

Comment:
Agreed, this could seem weird at first. However, after thinking about it a little bit more, I think there is a good justification for storing information in a non-User-model location. This flow is a somewhat self-contained product that we are hoping we will be able to get victim advocates to walk victims through on an iPad. The victims would receive an email confirmation, which then they could use to "create an account."

So long as we are using the terminology of "create an account" being something that the user needs to do, it seems conceptually clean to keep the information given prior to the user expressing their desire to create an account somewhere else.

I'm open to different opinions on this, though! Just figured that perhaps a bit more context about how we're thinking about this flow w/r/t the rest of the product will convince you.